runtime/dynlink.c

```diff
@@ -121,7 +121,7 @@ CAMLexport char_os * caml_parse_ld_conf(void)
       ("error while reading loader config file %s",
        caml_stat_strdup_of_os(ldconfname));
   config[nread] = 0;
-  wconfig = caml_stat_strdup_to_os(config);
+  wconfig = caml_stat_strdup_noexc_to_os(config);
   caml_stat_free(config);
   q = wconfig;
   for (p = wconfig; *p != 0; p++) {
```

Comment:
You need to check `wconfig` is not `NULL` now, since there will be no exception raised upon allocation failure.

Reply:
Actually, the whole premise of this was nonsense! I was under the (incorrect) impression that exceptions were not supposed to be raised during initialisation but that's wrong (cf #13516!). I'll revert this bit.